api/src/main/scala/hmda/api/model/WithPagination.scala

```diff
@@ -0,0 +1,79 @@
+package hmda.api.model
+
+trait WithPagination {
+  def pageSize: Int = 20
+
+  def path: String
+  def currentPage: Int
+  def total: Int
+
+  def links = PaginationLinks(
+    configurablePath,
+    pg(currentPage),
+    pg(1),
+    pg(prevPage),
+    pg(nextPage),
+    pg(lastPage)
+  )
+  def count: Int = {
+    if (validPage) {
+      if (currentPage == lastPage) lastPageCount
+      else pageSize
+    } else 0
+  }
+
+  ///// Helper Methods /////
+
+  private def validPage: Boolean = currentPage >= 1 && currentPage <= lastPage
+
+  private def configurablePath: String = s"$path{rel}"
+
+  private def lastPage: Int = {
+    if (total % pageSize == 0) fullPages
+    else fullPages + 1
+  }
+  private def prevPage: Int = {
+    if (currentPage < 2) 1
+    else currentPage - 1
+  }
+  private def nextPage: Int = {
+    if (currentPage >= lastPage - 1) lastPage
+    else currentPage + 1
+  }
+  private def fullPages: Int = total / pageSize
+
+  private def lastPageCount: Int = {
+    val remainder = total % pageSize
+    if (remainder == 0) pageSize
+    else remainder
+  }
+
+  private def pg(n: Int): String = s"?page=$n"
```

Comment:
Maybe `query` instead? I don't want to overload the word `page`, since it's used a lot of ways in this module.